demo/src/main/java/com/example/demo/service/PersonServiceImpl.java

```diff
@@ -51,9 +50,55 @@ public List<Person> findByEmail(String email) {
     }
 
     @Override
-    public List<Person> findByStreetAddress(String streetAddress) {
-        return personRepository.findByStreetAddress(streetAddress);
+    public List<Person> findByStreetAddress(String sAddress) {
+        return personRepository.findByStreetAddress(sAddress);
+    }
+
+    @Override
+    public Optional<Person> getPersonDetails(Long id) {
+        return personRepository.findWithAddressandConctact(id);
+    }
+
+    @Override
+    public Optional<Person> addAddressToPerson(Long personId, PostalAddress postalAddress) {
+        if (personId == null || postalAddress == null)
+            return Optional.empty();
+
+        Person person = personRepository.findById(personId).orElse(null);
+
+
+        assert person != null;
+        person.getAddresses().add(postalAddress);
+        this.personRepository.save(person);
+
+        if (!isAddressUniqueForPerson(personId, postalAddress.getStreetAddress())) {
+            return Optional.empty();
+        }
+        return Optional.of(person);
     }
 
+    private boolean isAddressUniqueForPerson(Long id, String address) {
+        return personRepository.countAddwithStreetAdd(id, address) == 0;
```

Comment:
No need to create separate method which checks if `streetAddress` already exists because in `PostalAddress.java` you have constraint (unique=true) for `streetAddress`